lib/neo4j/active_node/scope.rb

```diff
@@ -101,17 +101,31 @@ def initialize(target, query_proxy)
       end
 
       def identity
-        @query_proxy ? @query_proxy.identity : :n
+        query_proxy_or_target.identity
       end
 
       Neo4j::ActiveNode::Query::QueryProxy::METHODS.each do |method|
         module_eval(%{
-            def #{method}(params={})
+            def #{method}(*args)
               @target.all.scoping do
-                (@query_proxy || @target).#{method}(params)
+                query_proxy_or_target.#{method}(*args)
               end
             end}, __FILE__, __LINE__)
       end
```

Comment:
This might be a nice opportunity to transform this to `define_method` or `define_class_method` (whatever is appropriate).  Don't know how hard that would be